src/Nethermind/Nethermind.Consensus/Deposits/DepositsProcessor.cs

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Logging;
+using Nethermind.Serialization.Rlp;
+using Nethermind.State;
+using Nethermind.State.Proofs;
+
+namespace Nethermind.Consensus.Deposits;
+
+public class DepositsProcessor : IDepositsProcessor
+{
+    private readonly ILogger _logger;
+
+    public DepositsProcessor(ILogManager logManager)
+    {
+        ArgumentNullException.ThrowIfNull(logManager);
+
+        _logger = logManager.GetClassLogger();
+    }
+
+    public void ProcessDeposits(Block block, TxReceipt[] receipts, IReleaseSpec spec)
```

Comment:
Consider moving `IReleaseSpec` to ctor instead, or alternatively remove it and make the if check before consumers call `ProcessDeposits`.